incubator/etcd/templates/statefulset.yaml

```diff
@@ -34,141 +34,216 @@ spec:
             memory: "{{ .Values.Memory }}"
         env:
         - name: INITIAL_CLUSTER_SIZE
-          value: {{ default 3 .Values.Replicas | quote }}
+          value: "1"
         - name: SET_NAME
           value: {{ template "etcd.fullname" . }}
         volumeMounts:
         - name: datadir
-          mountPath: /var/run/etcd
+          mountPath: /etcd
         lifecycle:
           preStop:
             exec:
               command:
                 - "/bin/sh"
-                - "-ec"
+                - "-ecx"
                 - |
+                  # Cleaning up after a controlled stop, like a scale down of the cluster
+                  # is needed so the member is removed and could be started fresh
+                  {{- if .Values.PeerTLS }}
+                  PROTO=https
+                  {{- else }}
+                  PROTO=http
+                  {{- end }}
                   EPS=""
                   for i in $(seq 0 $((${INITIAL_CLUSTER_SIZE} - 1))); do
                       EPS="${EPS}${EPS:+,}http://${SET_NAME}-${i}.${SET_NAME}:2379"
                   done
 
                   HOSTNAME=$(hostname)
+                  ETCD_DIR="/etcd"
 
                   member_hash() {
-                      etcdctl member list | grep http://${HOSTNAME}.${SET_NAME}:2380 | cut -d':' -f1 | cut -d'[' -f1
+                      etcdctl member list | grep ${PROTO}://${HOSTNAME}.${SET_NAME}:2380 | cut -d':' -f1 | cut -d'[' -f1
                   }
 
                   echo "Removing ${HOSTNAME} from etcd cluster"
 
                   ETCDCTL_ENDPOINT=${EPS} etcdctl member remove $(member_hash)
                   if [ $? -eq 0 ]; then
                       # Remove everything otherwise the cluster will no longer scale-up
-                      rm -rf /var/run/etcd/*
+                      rm -rf $ETCD_DIR
                   fi
         command:
           - "/bin/sh"
-          - "-ec"
+          - "-ecx"
           - |
             HOSTNAME=$(hostname)
+            export ETCD_DATA_DIR="/etcd/default.etcd"
+            IP=$(ip addr | grep 'state UP' -A2 | tail -n1 | awk '{print $2}' | cut -f1  -d'/')
```

Comment:
This does not seem to work for minikube. The `grep` returns no matches resulting in an incomplete ` --listen-client-urls` on line 161.